xbmc/filesystem/CurlFile.cpp

```diff
@@ -601,9 +601,15 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
       StringUtils::Format(":%d", m_proxyport);
     g_curlInterface.easy_setopt(h, CURLOPT_PROXY, hostport.c_str());
 
-    const std::string userpass =
-      m_proxyuser + std::string(":") + m_proxypassword;
-    if (userpass != ":")
+    std::string userpass = "";
+
+    if (!m_proxyuser.empty() &&
+        !m_proxypassword.empty())
+      userpass = CURL::Encode(m_proxyuser) +
+          CURL::Encode(":") +
```

Comment:
not sure if this is needed, but might be bullet proof

Reply:
You want the ":" as plaintext.